rpc/rpcserver.go

```diff
@@ -47,19 +56,32 @@ func (r *rpcServer) Start(ctx context.Context, in *rpcapi.StartRequest) (*rpcapi
 		broadcastAcks = 1
 	}
 
+	gtwManager, err := gateway.NewManager(ctx, in.GatewayAddresses, connAcks)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { gtwManager.Close() }()
```

Comment:
These two versions do different things. According to https://go.dev/doc/effective_go#defer:
> The arguments to the deferred function (which include the receiver if the function is a method) are evaluated when the defer executes, not when the call executes.

I want to evaluate the _receiver_ later because it might be overwritten:
```go
	// Close the old connections and save the new manager.
	// The temporary manager is nil-ed to avoid closing connections in defer.
	r.gtwManager.Close()
	r.gtwManager, gtwManager = gtwManager, nil
```

Now when I look at it again, the snippet could also be just:
```go
	// Swap the new and old gateway managers. 
	// The old one will be closed in defer.
	r.gtwManager, gtwManager = gtwManager, r.gtwManager
```

Reply:
I forgot to remove this comment when I realised what is happening here. I would prefer not having a `defer` at all to call `gtwManager.Close()` since it makes the code unnecessarily hard to follow.

Reply:
Defer prevents leaking resources, so I'd rather keep it.

Reply:
I don't agree that it's dangerous, nothing bad can happen. 

The additional `nil` check would need to be present anyways. Using `defer` is not the source of it. It's required because initially `r.gtwManager` might be `nil` (if there weren't enough successful gtw connections when the server was started). I can get rid of it by placing an "empty" `gateway.Manager` in this case.